src/lib/provable/gadgets/twisted-curve.ts

```diff
@@ -0,0 +1,462 @@
+import { inverse, mod } from '../../../bindings/crypto/finite-field.js';
+import { Provable } from '../provable.js';
+import { assert } from './common.js';
+import { Field3, ForeignField, split } from './foreign-field.js';
+import { l2Mask } from './range-check.js';
+import { sha256 } from 'js-sha256';
+import {
+  bigIntToBytes,
+  bytesToBigInt,
+} from '../../../bindings/crypto/bigint-helpers.js';
+import {
+  CurveTwisted,
+  GroupTwisted,
+  twistedAdd,
+  twistedDouble,
+} from '../../../bindings/crypto/elliptic-curve.js';
+import { assertPositiveInteger } from '../../../bindings/crypto/non-negative.js';
+import { sliceField3 } from './bit-slices.js';
+import { exists } from '../core/exists.js';
+import { arrayGetGeneric, point, Point } from './elliptic-curve.js';
+
+// external API
+export { EllipticCurveTwisted };
+
+const EllipticCurveTwisted = {
+  add,
+  double,
+  negate,
+  assertOnCurve,
+};
+
+function add(
+  p1: Point,
+  p2: Point,
+  Curve: { modulus: bigint; a: bigint; d: bigint }
```

Comment:
Is `modulus` the order of the curve? or the modulus of the base field?

Reply:
the order of the base foreign field I guess, based on the code below.

Reply:
the base field yes